upload/common/uploaded_file.py

```diff
@@ -82,11 +84,16 @@ def size(self):
         else:
             return self.s3obj.content_length
 
+    @retry(wait=wait_fixed(2), stop=stop_after_attempt(5))
```

Comment:
do you want to update the test around this? Something like Sam's retry test for staged files https://github.com/HumanCellAtlas/upload-service/commit/ee79ccb8f1196e2cfd5f6541dd711b7f3bc059ff#diff-a0b391d1dadf36d54aa749d1fc35ffa5R55

Reply:
Hmmm, what is the value add of extensive mocking to test imported retry functionality? We are using a version of tenacity with a passing build. If you went down this path, you would have multiple additional tests for every single instance of retry in our codebase. Personally, I think this quickly tilts in the direction of tests losing their value

Reply:
This is a tricky balance.

On the one hand, yes, we don't need to test the Retry library.  On the other hand, you do want to check that the condition that *should* cause a retry *does* cause a retry.  So you're not writing tests for the `@retry` code so much as the `if tags are missing raise` code.

That said, it is expensive to develop such a test (or at least it was last time).  The mocking required is a little intricate and it requires you to divide your code into two functions, one that has an `@retry` and one that you can mock out to cause the desired retry.

So I'd not call this a mandatory requirement in all cases.  If you have the time to test it, do so.  If it is very complex code and hard to tell if the code path will be executed correctly, test it.  But if it is trivial to visually inspect, and too expensive to test, given the other 99 things on your plate, skip it.